docs/decisions/00NN-breaking-changes-guidance.md

```diff
@@ -0,0 +1,41 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: proposed
+contact: markwallace
+date: 2024-06-05
+deciders: sergeymenshykh, mbolan, rbarreto, dmytrostruk, westey
+consulted: 
+informed: 
+---
+
+# Guidance for Breaking Changes
+
+## Context and Problem Statement
+
+We must avoid breaking changes in .Net because of the well known [diamond dependency issue](https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/dependencies#diamond-dependencies) where breaking changes between different versions of the same package cause bugs and exceptions at run time.
+
+## Decision Drivers
+
+Breaking changes are only allowed under the following circumstances:
+
+- Updates to an experimental feature i.e. we have learnt something new and need to modify the design of an experimental feature.
+- When one of our dependencies introduces an unavoidable breaking change.
+
+All breaking changes must be clearly documented, definitely in the release notes and possibly also via a migration guide Blog post.
```

Comment:
The migration guide should probably be in our learn site instead of the blog so it can be translated.